server/clustering.go

```diff
@@ -47,6 +51,9 @@ func (r *raftNode) shutdown() error {
 	if err := r.transport.Close(); err != nil {
 		return err
 	}
+	if err := r.joinSub.Unsubscribe(); err != nil {
```

Comment:
I see that we create the `transport` based on `s.ncr`. I assume `r.transport.Close()` is not closing the underlying NATS Connection, right? If so, the unsubscribe here would fail.

Reply:
Your assumption is correct. Transport close here just shuts down the Raft transport but not the underlying NATS conn.